app/services/form_service.py

```diff
@@ -1,80 +1,102 @@
-                    return ResponseDTO(FormModel.helper(new_form),"Form created successfully", http.HTTPStatus.CREATED)
+                    #new_form = self.form_collection.find_one(result.inserted_id)
+                    return ResponseDTO(result.inserted_id,"Form created successfully", http.HTTPStatus.CREATED)
                 else:
                     return ResponseDTO(None, "Couldn't Create Form. Aborting.", http.HTTPStatus.BAD_REQUEST)
         except Exception as e:
             # TODO:Utilizar a biblioteca logging para criar uma documentação clara do que esta rolando na api. Nota: Isso facilita o debug e rastreabilidade tmb
             print(f"Error creating Forms: {e}")
             return ResponseDTO(None, "Error Creating Form: " + str(e), http.HTTPStatus.BAD_REQUEST)
     
-    def get_form_by_id(self, ong_id: str,  animal_id:str, form_id: str) -> ResponseDTO:
+    def get_form_by_id(self, ong_id: str,  animal_id:str, form_id: str, request_id: str = "") -> ResponseDTO:
+        self.logger.info(f"id = {request_id} starting get form by id")
         try:
             response = self.ong_service.get_ong_animals(ong_id)
             if response.status != http.HTTPStatus.OK:
                 return response
             animals = response.data
-
-            has_animal = False
+            if not animals:
+                 return ResponseDTO(None, "There is no animals in ONG.", http.HTTPStatus.BAD_REQUEST)
+            animal_exist = False
             animal_object = None
-            has_form = False
+            form_exist = False
             for animal in animals:
                 if str(animal.get("id")) == animal_id:
-                    has_animal = True
+                    animal_exist = True
                     animal_object = animal
                     break
-            if not has_animal:
+            if not animal_exist:
                 return ResponseDTO(None, "Animal doesn't belongs to ONG.", http.HTTPStatus.BAD_REQUEST)
             
-            for f in animal_object.get("forms"):
-                if f == form_id:
-                    has_form = True
-            
-            if not has_form:
+            for form in animal_object.get("forms"):
+                if form == form_id:
+                    form_exist = True
+                    break
+                
+            if not form_exist:
                 return ResponseDTO(None, "This form doesn't belong to this animal", http.HTTPStatus.BAD_REQUEST)
+            return self.get_form(form_id, request_id)
+            
+        except Exception as e:
+            message = f"Error getting Form: {e}"
+            print(message)
+            return ResponseDTO(None, message, http.HTTPStatus.BAD_REQUEST)
+    
+    def get_form (self, form_id: str, request_id: str = "")-> ResponseDTO:
+        if request_id == "":
+            self.logger.info(f"id = {request_id} starting get form")
```

Comment:
esse request_id precisa ser passado em todas as funções, ver com calma
```suggestion
        self.logger.info(f"id = {request_id} starting get form")
```